### PR TITLE
docs/cuda: reword cuda-aware support of communication APIs

### PR DESCRIPTION
**Pull request by wenduwan**
This change also reorganizes the paragraphs and removes duplicate contents.

For testing, I did a local `make dist` and inspected the html file.

### CODE REVIEW

File: docs/tuning-apps/networking/cuda.rst [hunk 1]
```diff
@@ -410,135 +286,75 @@ example of how to write CUDA-aware MPI applications.
...
-* MPI_Iexscan
-
-.. FIXME: We need to verify the above list.  These _SHOULD_ be the same
-   as above.
-
-/////////////////////////////////////////////////////////////////////////
 
-Which MPI APIs do NOT work with CUDA-aware UCX?
------------------------------------------------
+However, the following APIs do not support GPU buffers:
```
Comment (edgargabriel) [inline docs/tuning-apps/networking/cuda.rst:357]: is this really correct?  at least with recent UCX versions I think some/most of these functions should work.

**Review by edgargabriel — CHANGES_REQUESTED** [review]

File: docs/tuning-apps/networking/cuda.rst [hunk 2]
```diff
@@ -4,51 +4,96 @@ CUDA
...
+* Both CUDA-ized shared memory (``smcuda``) and TCP (``tcp``) BTLs
+  with the OB1 (``ob1``) PML.
+* The HCOLL (``hcoll``) COLL
 
-   It should look something like:
+See :ref:`this FAQ entry <faq-cuda-mpi-apis-cuda-label>`
+for more details on which commnication APIs are CUDA-aware.
 
-   .. code-block:: sh
+OFI support for CUDA
```
Comment (hppritcha) [inline docs/tuning-apps/networking/cuda.rst:58]: Might want to add something about versions of libfabric to use.  Maybe say something like latest version of libfabric is recommended or at least something newer than 1.9?

**Review by bosilca — APPROVED** [review]

File: docs/tuning-apps/networking/cuda.rst (hunk 1 above)
Comment (wenduwan) [reply docs/tuning-apps/networking/cuda.rst:357]: @bosilca @janjust Do you know if UCC currently supports non-blocking collectives with CUDA buffer? I tried single-node and from what I saw it worked.

Also could you help double check the statement about UCX CUDA support status?

Comment (janjust) [reply docs/tuning-apps/networking/cuda.rst:357]: UCC supports both blocking and non-blocking collectives with cuda buffers

Comment (janjust) [reply docs/tuning-apps/networking/cuda.rst:357]: UCX supports cuda buffers I think for most, not sure about cuda-ipc but osc ucx does not I think. I have to double check but I'm pretty sure we have issues in accumulate

Comment (wenduwan) [reply docs/tuning-apps/networking/cuda.rst:357]: Thanks Tommy! I will update the UCC part first.

**Comment by wenduwan** [comment]
Sorry I won't have time to work on this. Closing...

**Pull request closed without merging**